src/enemy.rs

```diff
@@ -187,16 +186,16 @@ fn spawn(
     mut spawn_map: ResMut<EnemySpawners>,
     mut player: Query<&mut Transform, With<Player>>,
     mut current_wave: ResMut<CurrentWave>,
-    mut global_time_ticker: Res<GlobalTimeTickerResource>,
+    global_time_ticker: Res<GlobalTimeTickerResource>,
 ) {
     if global_time_ticker.0.finished() || current_wave.need_wave_spawn {
         if let Some(spawners) = spawn_map.0.get_mut(&current_wave.num) {
-            for mut spawner in spawners {
+            for spawner in spawners {
                 spawner.timer.tick(global_time_ticker.0.duration());
 
                 if spawner.timer.finished() || current_wave.need_wave_spawn {
                     // spawn elite only one time on the wave
-                    if Some(true) == spawner.is_elite && current_wave.need_wave_spawn == false {
+                    if Some(true) == spawner.is_elite && !current_wave.need_wave_spawn {
```

Comment:
It doesn't make much sense for this field to be `Option`. IIRC, it is only intended to deserialize a configuration with a missing `is_elite` field. In this case, we could use `#[serde(default = "false")] `instead